### ISSUE TITLE
Use the locale in String.formta()

### ISSUE THREAD

**Issue by armandomonteiroptc** [DiUS/java-faker#441]
**Describe the bug**
Formatting string should benefit from the Faker (Locale.ENGLISH by default), otherwise a formatted string like the longitude or latitude, can be **_1.234567_** or **_1,234567_** depending in the Locale.

**To Reproduce**

Using a `String.format("%.8g", (faker.random().nextDouble() * 180) - 90);` 
* depends on external JVM configuration 
  * if Locale PT then **_1,234567_**.

Using a `String.format(Locale.ENGLISH, "%.8g", (faker.random().nextDouble() * 180) - 90);` 
* controlled by the application
  * then **_1.234567_**.

[String.format()](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#format-java.util.Locale-java.lang.String-java.lang.Object...-)

**Expected behavior**
`String.format(Locale.ENGLISH, "%.8g", (faker.random().nextDouble() * 180) - 90);`
* controlled by the application
  * then **_1.234567_**.

**Versions:**
 - OS: Any
 - JDK: Any
 - Faker Version: using 0.18 - however the implementation seems to be the same in the last version

**Additional context**
-

**Comment by maxtar** [comment]
Yes we have the same problem. Locale ignored for String.format().

**Comment by AlexZhuk** [comment]
Same for price() method in Commerce class

`
public String price(double min, double max) {
        double price = min + this.faker.random().nextDouble() * (max - min);
        return (new DecimalFormat("#0.00")).format(price);
    }
`
as wee can see DecimalFormat is used without locale.
Should be something like : 
`
new DecimalFormat("#0.00", new DecimalFormatSymbols(locale)).format(price);

`
where locale should be used default system (if not stated), otherwise use stated locale
